graphrag/vector_stores/couchbasedb.py

```diff
@@ -0,0 +1,193 @@
+    """The Couchbase vector storage implementation."""
+
+    def __init__(
+        self,
+        collection_name: str,
+        bucket_name: str,
+        scope_name: str,
+        index_name: str,
+        text_key: str = "text",
+        embedding_key: str = "embedding",
+        scoped_index: bool = True,
+        **kwargs: Any,
+    ):
+        super().__init__(collection_name, **kwargs)
+        self.bucket_name = bucket_name
+        self.scope_name = scope_name
+        self.index_name = index_name
+        self.text_key = text_key
+        self.embedding_key = embedding_key
+        self.scoped_index = scoped_index
+        self.vector_size = kwargs.get("vector_size", DEFAULT_VECTOR_SIZE)
+        logger.debug(
+            f"Initialized CouchbaseVectorStore with collection: {collection_name}, bucket: {bucket_name}, scope: {scope_name}, index: {index_name}"
+        )
+
+    def connect(self, **kwargs: Any) -> None:
+        """Connect to the Couchbase vector store."""
+        connection_string = kwargs.get("connection_string")
+        username = kwargs.get("username")
+        password = kwargs.get("password")
+
+        if not isinstance(username, str) or not isinstance(password, str):
+            logger.error("Username and password must be strings")
+            raise TypeError("Username and password must be strings")
+        if not isinstance(connection_string, str):
+            logger.error("Connection string must be a string")
+            raise TypeError("Connection string must be a string")
+
+        logger.info(f"Connecting to Couchbase at {connection_string}")
+        auth = PasswordAuthenticator(username, password)
+        options = ClusterOptions(auth)
+        cluster = Cluster(connection_string, options)
+        self.db_connection = cluster
+        self.bucket = cluster.bucket(self.bucket_name)
+        self.scope = self.bucket.scope(self.scope_name)
+        self.document_collection = self.scope.collection(self.collection_name)
+        logger.info("Successfully connected to Couchbase")
+
+    def load_documents(
+        self, documents: list[VectorStoreDocument], overwrite: bool = True
+    ) -> None:
+        """Load documents into vector storage."""
+        logger.info(f"Loading {len(documents)} documents into vector storage")
+        batch = [
+            {
+                "id": doc.id,
+                self.text_key: doc.text,
+                self.embedding_key: doc.vector,
+                "attributes": json.dumps(doc.attributes),
+            }
+            for doc in documents
+            if doc.vector is not None
+        ]
+        if batch:
+            successful_loads = 0
+            for doc in batch:
+                try:
+                    if overwrite:
+                        self.document_collection.upsert(doc["id"], doc)
+                    else:
+                        self.document_collection.insert(doc["id"], doc)
+                    successful_loads += 1
```

Comment:
Use upsert_multi in batch.
Also you don't need different upsert and insert in case of overwrite, Couchbase will handle it itself based on key

Reply:
used upsert_multi now